src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -847,7 +847,8 @@ def write_string_to_logfile(self, message):
             cart_ctl.add_log_msg.value = "add_log_msg"
             cart_ctl.rank.value = "all"
             cart_ctl.log_message.value = message
-            cart_ctl.no_sync.value = None
+            # Don't ping all ranks before sending the log command
+            cart_ctl.no_sync.value = True
```

Comment:
this part is technically 'the fix'. the rest of changes are additions for other situations